data_steward/cdr_cleaner/cleaning_rules/recent_concept_suppression.py

```diff
@@ -0,0 +1,238 @@
+            primary_date_field = get_primary_date_field(table_name)
+
+            if concept_id_fields and primary_date_field and table_name in tables_in_dataset:
+                table_infos.append({
+                    'table_name': table_name,
+                    'primary_key': primary_key,
+                    'primary_datefield': primary_date_field,
+                    'concept_id_fields': concept_id_fields
+                })
+
+        retrieve_all_concepts_first_uses = RETRIEVE_CONCEPT_FIRST_USES_QUERY.render(
+            table_infos=table_infos,
+            project_id=self.project_id,
+            dataset_id=self.dataset_id,
+            sandbox_id=self.sandbox_dataset_id,
+            concept_first_use_table=CONCEPT_FIRST_USE_TABLE,
+            default_valid_start_date=DEFAULT_CONCEPT_VALID_START_DATE)
+
+        query_job = client.query(retrieve_all_concepts_first_uses)
+        result = query_job.result()
+
+        if hasattr(result, 'errors') and result.errors:
+            LOGGER.error(f"Error running job {result.job_id}: {result.errors}")
+            raise GoogleCloudError(
+                f"Error running job {result.job_id}: {result.errors}")
+
+    def validate_rule(self, client, *args, **keyword_args):
+        """
+        Validates the cleaning rule which deletes or updates the data from the tables
+
+        Method to run validation on cleaning rules that will be updating the values.
+        For example:
+        if your class updates all the datetime fields you should be implementing the
+        validation that checks if the date time values that needs to be updated no
+        longer exists in the table.
+
+        if your class deletes a subset of rows in the tables you should be implementing
+        the validation that checks if the count of final final row counts + deleted rows
+        should equals to initial row counts of the affected tables.
+
+        Raises RunTimeError if the validation fails.
+        """
+
+        raise NotImplementedError("Please fix me.")
+
+    def setup_validation(self, client, *args, **keyword_args):
+        """
+        Run required steps for validation setup
+
+        Method to run to setup validation on cleaning rules that will be updating or deleting the values.
+        For example:
+        if your class updates all the datetime fields you should be implementing the
+        logic to get the initial list of values which adhere to a condition we are looking for.
+
+        if your class deletes a subset of rows in the tables you should be implementing
+        the logic to get the row counts of the tables prior to applying cleaning rule
+
+        """
+        raise NotImplementedError("Please fix me.")
+
+
+if __name__ == '__main__':
+    import cdr_cleaner.args_parser as parser
+    import cdr_cleaner.clean_cdr_engine as clean_engine
+
+    ARGS = parser.parse_args()
+    pipeline_logging.configure(level=logging.DEBUG, add_console_handler=True)
```

Comment:
You should add the `cutoff_date` argument here so we can list the queries from the command line.  The following raises an error:

```shell
lbiggers $ PYTHONPATH=./:$PYTHONPATH python cdr_cleaner/cleaning_rules/recent_concept_suppression.py -p project_foo -d dataset_bar -b sandbox_baz -l  --cutoff_date 2022-07-01
usage: recent_concept_suppression.py [-h] -p PROJECT_ID -d DATASET_ID -b SANDBOX_DATASET_ID [-s] [-l]
recent_concept_suppression.py: error: unrecognized arguments: --cutoff_date 2022-07-01
```

Reply:
Sounds good, I'll add all of these suggestions.